packages/@react-aria/calendar/src/utils.ts

```diff
@@ -67,10 +76,43 @@ export function useVisibleRangeDescription(startDate: CalendarDate, endDate: Cal
       if (isSameDay(endDate, endOfMonth(startDate))) {
         return monthFormatter.format(startDate.toDate(timeZone));
       } else if (isSameDay(endDate, endOfMonth(endDate))) {
-        return monthFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
+        return isAria
+          ? formatRange(monthFormatter, formatMessage, startDate, endDate, timeZone)
+          : monthFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
       }
     }
 
-    return dateFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
-  }, [startDate, endDate, monthFormatter, dateFormatter, timeZone]);
+    return isAria
+      ? formatRange(dateFormatter, formatMessage, startDate, endDate, timeZone)
+      : dateFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
+  }, [startDate, endDate, monthFormatter, dateFormatter, formatMessage, timeZone, isAria]);
+}
+
+function formatRange(dateFormatter: DateFormatter, formatMessage: FormatMessage, start: CalendarDate, end: CalendarDate, timeZone: string) {
+  let parts = dateFormatter.formatRangeToParts(start.toDate(timeZone), end.toDate(timeZone));
+
+  // Find the separator between the start and end date. This is determined
+  // by finding the last shared literal before the end range.
+  let separatorIndex = -1;
```

Comment:
Yep, see https://github.com/adobe/react-spectrum/pull/3005#discussion_r842044593. Any ideas where to put it?

Reply:
gotcha, well, dependency chain would dictate that it lives in stately?

Reply:
Which stately package though? This is shared between `@react-aria/calendar` and `@react-aria/datepicker` - there's no common stately package...

Reply:
wait, so it's in three places? the comment you linked to said it's in react-stately/datepicker?

so react-aria/datepicker uses the types from react-types/calendar & react-types/datepicker, and uses react-stately/datepicker
and react-aria/calendar uses tye types from react-types/calendar, and uses react-stately/calendar

what's interesting about that though is that
react-stately/calendar uses the types from react-types/calendar & react-types/datepicker
and react-stately/datepicker uses the types from react-types/datepicker

so I think the most expected place for it would be a new package... or for one of these to be more clearly the *base* of the dependencies?

Reply:
no just two, stately/datepicker and aria/calendar. I think it would be weird for aria/calendar to depend on stately/datepicker. Ugh, not sure about a whole new package though.

Reply:
we can decide later